### PR TITLE
Add passive validation - list local group memberships

### PR DESCRIPTION
**Pull request by max-zaytsev**
Added local group membership list for the current user

When the pipeline is running with diagnostics, this information will be added to the environment.txt file

For example:
```
Local group membership for current user: REDMOND\v-mazayt
docker-users
Administrators
Performance Log Users
```

### CODE REVIEW

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 1]
```diff
@@ -328,6 +336,53 @@ private async Task<string> GetPsVersionInfo()
             return builder.ToString();
         }
 
+        private async Task<string> GetLocalGroupMembership(string userName)
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
+            string arguments = $@"
+foreach ($group in Get-LocalGroup ) {{
```
Comment (anatolybolshakov) [inline src/Agent.Worker/DiagnosticLogManager.cs:345]: Could it make sense to move it to the separate file?

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 2]
```diff
@@ -328,6 +336,53 @@ private async Task<string> GetPsVersionInfo()
             return builder.ToString();
         }
 
+        private async Task<string> GetLocalGroupMembership(string userName)
```
Comment (anatolybolshakov) [inline src/Agent.Worker/DiagnosticLogManager.cs:339]: Please add /// comments

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 3]
```diff
@@ -328,6 +336,53 @@ private async Task<string> GetPsVersionInfo()
...
+        {
+            var builder = new StringBuilder();
+
+            string powerShellExe = HostContext.GetService<IPowerShellExeUtil>().GetPath();
```
Comment (anatolybolshakov) [inline src/Agent.Worker/DiagnosticLogManager.cs:343]: Do we probably need some fallback to previous behavior with try/catch block - so if there is an error, it won't crush agent, but just write some message in logs about the exception?

File: src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1 [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    $groupMembers = @($g.Invoke('Members') | ForEach-Object { ([adsi]$_).path }) 
+    $names = foreach ($member in $groupMembers) {
+        $x = [regex]::match($member, '^WinNT://(.*)').groups[1].value;
+        $x.Replace("`/", "`\");
+    }
+    if ($names -contains $UserName) {
+        $true
+    } else {
+        $false
+    }
```
Comment (alexander-smolyakov) [inline src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1:23]:
```suggestion
    return ($names -contains $UserName)
```

File: src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1 [hunk 5]
```diff
@@ -0,0 +1,50 @@
+[CmdletBinding()]
+param()
+
+# Checks if a user is a member of a group using ADSI
```
Comment (alexander-smolyakov) [inline src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1:4]: Could please describe in more detail how this test works?

File: src/Agent.Worker/DiagnosticLogManager.cs [hunk 6]
```diff
@@ -357,6 +357,12 @@ private async Task<string> GetEnvironmentContentWindows(int agentId, string agen
             // $psversiontable
             builder.AppendLine("Powershell Version Info:");
             builder.AppendLine(await GetPsVersionInfo());
+
+            if(PlatformUtil.RunningOnWindows)
```
Comment (AndreyIvanov42) [inline src/Agent.Worker/DiagnosticLogManager.cs:361]: There is a similar check above (line 330)

**Review by alexander-smolyakov — COMMENTED** [review]
Please take a look at the comments

File: src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1 (hunk 5 above)
Comment (max-zaytsev) [reply src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1:4]: This is not a test :) I just used the [approved](https://docs.microsoft.com/en-us/powershell/scripting/developer/cmdlet/approved-verbs-for-windows-powershell-commands?view=powershell-7.2) verb for naming. I can rename it if needed

Comment (alexander-smolyakov) [reply src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1:4]: @max-zaytsev I mean, could you please add some code comments to describe how the function works?

File: src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1 [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+function Test-LocalGroupMembershipADSI {
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)]
+        [string]$Group,
+        [Parameter(Mandatory = $true)]
+        [string]$UserName
+    )
+    
+    # Get a group object using ADSI adpater
```
Comment (anatolybolshakov) [inline src/Misc/layoutbin/powershell/Get-LocalGroupMembership.ps1:15]:
```suggestion
    # Get a group object using ADSI adapter
```

**Review by anatolybolshakov — APPROVED** [review]
LGTM, thanks!

**Review by alexander-smolyakov — APPROVED** [review]

**Review by DaniilShmelev — APPROVED** [review]
LGTM! Also tested these changes manually and found no issues.

**Pull request merged**